rotkehlchen/exchanges/manager.py

```diff
@@ -82,11 +83,20 @@ def get_exchange(self, name: str, location: Location) -> Optional[ExchangeInterf
 
         return None
 
+    def get_excluded_exchanges(self) -> Iterable[ExchangeLocationID]:
+        if not hasattr(self, 'database'):
+            return []
+        db_settings = self.database.get_settings()
+        return db_settings.non_syncing_exchanges
+
     def iterate_exchanges(self) -> Iterator[ExchangeInterface]:
         """Iterate all connected exchanges"""
+        excluded = set(self.get_excluded_exchanges())
         for _, exchanges in self.connected_exchanges.items():
             for exchange in exchanges:
-                yield exchange
+                # We are not yielding excluded exchanges
+                if exchange.location_id() not in excluded:
```

Comment:
@LefterisJP  About an iteration function I've had a call with Yabir where he said that modification of this function would be enough to solve the issue.
About skipping direct queries I think that it will be a good feature for user but I'm a bit afraid that it could be difficult to create a reliable and clever enough skipping algorithm and test it.

Reply:
Sure let's keep it like that for now and we will see. This is not really a big priority feature and if we see we need changes asked by the user we can implement them then.